packages/algob/src/lib/tx.ts

```diff
@@ -189,49 +192,43 @@ function signTransaction (txn: Transaction, txnParam: execParams): Uint8Array {
 /**
  * Description: send signed transaction to network and wait for confirmation
  * @param deployer AlgobDeployer
- * @param txns Signed Transaction(s)
+ * @param rawTxns Signed Transaction(s)
  */
 async function sendAndWait (
   deployer: AlgobDeployer,
-  txns: Uint8Array | Uint8Array[]): Promise<tx.ConfirmedTxInfo> {
-  const txInfo = await deployer.algodClient.sendRawTransaction(txns).do();
+  rawTxns: Uint8Array | Uint8Array[]): Promise<algosdk.ConfirmedTxInfo> {
+  const txInfo = await deployer.algodClient.sendRawTransaction(rawTxns).do();
   return await deployer.waitForConfirmation(txInfo.txId);
 }
 
 export async function executeTransaction (
   deployer: AlgobDeployer,
-  txnParams: execParams | execParams[]): Promise<tx.ConfirmedTxInfo> {
-  if (Array.isArray(txnParams)) {
-    if (txnParams.length > 16) {
-      throw new Error("Maximum size of an atomic transfer group is 16");
-    }
-
-    const txns = [];
-    for (const txnParam of txnParams) {
-      const suggestedParams = await mkSuggestedParams(deployer.algodClient, txnParam.payFlags);
-      const txn = mkTransaction(txnParam, suggestedParams);
-      txns.push(txn);
-    }
-    tx.assignGroupID(txns);
-
-    const signedTxns = [] as Uint8Array[];
-    txns.forEach((txn, index) => {
-      const signed = signTransaction(txn, txnParams[index]);
-      deployer.log(`Transaction ${index}`, signed);
-      signedTxns.push(signed);
+  execParams: ExecParams | ExecParams[]): Promise<algosdk.ConfirmedTxInfo> {
+  const suggestedParams = await getSuggestedParams(deployer.algodClient);
+  const mkTx = async (p: ExecParams): Promise<Transaction> =>
+    mkTransaction(p,
+      await mkTxParams(deployer.algodClient, p.payFlags, Object.assign({}, suggestedParams)));
+
+  let signedTxn;
+  if (Array.isArray(execParams)) {
+    if (execParams.length > 16) { throw new Error("Maximum size of an atomic transfer group is 16"); }
+
+    let txns = [];
+    for (const t of execParams) { txns.push(await mkTx(t)); }
+    txns = algosdk.assignGroupID(txns);
+    signedTxn = txns.map((txn: Transaction, index: number) => {
```

Comment:
I am curious about this. does this creates a map and stores it in signedTxn ? How is it different than using `forEach` ?

Reply:
`map` returns a list. `forEach` only iterates
explanation of map:
```
let L = some_list;
L.map(f) == [f(L[0]), f(L[1]), .... f(L[n])]
```